internal/kafka/internal/handlers/validation.go

```diff
@@ -38,8 +38,14 @@ const minimunNumberOfNodesForTheKafkaMachinePool = 3
 func validateKafkaBillingModel(ctx context.Context, kafkaService services.KafkaService, kafkaConfig *config.KafkaConfig, kafkaRequestPayload *public.KafkaRequestPayload) handlers.Validate {
 	return func() *errors.ServiceError {
 		billingModel := shared.SafeString(kafkaRequestPayload.BillingModel)
-		// enterprise kafkas billing model validation
 
+		// No explicitly set kafka billing mode is allowed for now, in which case
+		// an implementation-defined default is chosen
+		if shared.StringEmpty(billingModel) {
```

Comment:
this moves the condition from after to before the billing model check. Where does enterprise get assigned?

Reply:
It's assigned in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/c18897b7b796cc1e1029659665314c195c01e3df/internal/kafka/internal/presenters/kafka.go#L46